### DISCUSSION TITLE
关于Web反向代理Xray服务

### DISCUSSION THREAD

**Discussion by bigbugcc** [XTLS/Xray-core#533] [Q&A] [answered]
服务器上有一个网站，想在这个网站下的/xray路径反向代理Xray服务(说直白点也就是V2ray ws + tls那套伪装玩法)，参考了很多配置还是不行，提示400报错。
yun.abc.com是一个网站

`
{
  "log": {
    "access": "/var/log/xray/access.log",
    "error": "/var/log/xray/error.log",
    "loglevel": "info"
  },
  "inbounds": [
    {
      "port": 1129,
      "protocol": "vless",
      "settings": {
        "clients": [
          {
            "id": "d3f16e08-a24d-11eb-a1a3-e3ddb3208ddd"
          }
        ],
        "decryption": "none"
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls",
        "tlsSettings": {
          "certificates": [
            {
              "certificateFile": "/www/server/panel/vhost/ssl/yun.abc.com/fullchain.pem",
              "keyFile": "/www/server/panel/vhost/ssl/yun.abc.com/privkey.pem"
            }
          ],
          "alpn": [
            "http/1.1"
          ]
        },
        "tcpSettings": {},
        "kcpSettings": {},
        "httpSettings": {},
        "wsSettings": {
          "path": "/sync",
          "headers": {
            "Host": "yun.abc.com"
          }
        },
        "quicSettings": {}
      },
      "domain": "yun.abc.com"
    }
  ],
  "outbounds": [
    {
      "protocol": "freedom",
      "settings": {}
    },
    {
      "protocol": "blackhole",
      "settings": {},
      "tag": "blocked"
    }
  ],
  "routing": {
    "rules": [
      {
        "type": "field",
        "ip": [
          "0.0.0.0/8",
          "10.0.0.0/8",
          "100.64.0.0/10",
          "169.254.0.0/16",
          "172.16.0.0/12",
          "192.0.0.0/24",
          "192.0.2.0/24",
          "192.168.0.0/16",
          "198.18.0.0/15",
          "198.51.100.0/24",
          "203.0.113.0/24",
          "::1/128",
          "fc00::/7",
          "fe80::/10"
        ],
        "outboundTag": "blocked"
      }
    ]
  }
}
`

nginx的反向代理配置
`
location /sync
{
    proxy_pass http://127.0.0.1:1129;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header REMOTE-HOST $remote_addr;
    
    add_header X-Cache $upstream_cache_status;
    
    #Set Nginx Cache
    
    	add_header Cache-Control no-cache;
}
`

错误提示 
v2ray.com/core/transport/internet/websocket: failed to dial WebSocket > v2ray.com/core/transport/internet/websocket: failed to dial to (wss://yun.abc.com/sync):  > dial tcp: operation was canceled] > v2ray.com/core/common/retry: all retry attempts failed

**Comment by xqzr** [comment, marked as answer]
在 `nginx` 上配置SSL证书，而不是在 `xray` 上
